lumicks/pylake/kymotracker/kymotracker.py

```diff
@@ -377,6 +383,38 @@ def refine_tracks_centroid(tracks, track_width=None):
     track_width_pixels = np.ceil(track_width / tracks._kymo.pixelsize[0])
 
     interpolated_tracks = [track.interpolate() for track in tracks]
+    image = tracks._kymo.get_image(tracks._channel)
+
+    if pixel_threshold is not None:
+        if pixel_threshold <= 0:
+            raise ValueError(f"pixel_threshold should be larger than zero")
+
+        blurred = gaussian_filter(image, [0.5, 0])
+        kymo = tracks._kymo
+        blurred_kymo = _kymo_from_array(
+            blurred,
+            tracks._channel[0],
+            kymo.line_time_seconds,
+            start=kymo.start,
+            pixel_size_um=kymo.pixelsize_um[0],
+        )
+
+        blurred_tracks = [
+            KymoTrack(track.time_idx, track.coordinate_idx, blurred_kymo, tracks._channel)
+            for track in interpolated_tracks
+        ]
+
+        sample_window = int(track_width_pixels / 2)
+        keep_nodes = [
+            np.array(track.sample_from_image(sample_window, np.max)) >= pixel_threshold
+            for track in blurred_tracks
+        ]
```

Comment:
To me, this seems a bit smelly and it feels like there should be a low level API that just does this low level.

Reply:
The feature itself seems to work correctly (tested it).

Reply:
I tried it as well, and it changes some tracks even when I did no stitching operation.

```python
kymo_ds = kymo.downsampled_by(time_factor=2)
lines = lk.track_greedy(kymo_ds, "green", **settings)
refined_lines = lk.refine_tracks_centroid(
  lines,
  track_width=settings["track_width"],
  pixel_threshold=settings["pixel_threshold"]
)  # AFAIK, this line should've been a nullop but it's not.
```

Yellow is after refinement.
![image](https://user-images.githubusercontent.com/19836026/191461633-bd50d96a-9770-4d66-84b7-e69baa9b941d.png)

Reply:
> To me, this seems a bit smelly and it feels like there should be a low level API that just does this low level.

yeah you're right, pulled the image sampling stuff out to a separate function that returns the sub-array basically and left the `reduce` in the method. thoughts?